api/v1beta1/openstacknetconfig_webhook.go

```diff
@@ -247,14 +275,79 @@ func (r *OpenStackNetConfig) validateStaticMacReservations(old runtime.Object) e
 				if oldInstance, ok = old.(*OpenStackNetConfig); !ok {
 					return fmt.Errorf("runtime object is not an OpenStackNetConfig")
 				}
-				if currentMAC, ok := oldInstance.Spec.OVNBridgeMacMappings.StaticReservations[node].Reservations[physnet]; ok && currentMAC != mac {
+				if currentMAC, ok := oldInstance.Spec.Reservations[node].MACReservations[physnet]; ok && currentMAC != mac {
 					return fmt.Errorf("MAC address %s of node %s must not change - new MAC address %s", currentMAC, node, mac)
 				}
 			}
 		}
 
 		// if all tests pass add to reservations
-		reservations[node] = res
+		reservations[node] = OpenStackMACNodeReservation{
+			Reservations: res.MACReservations,
+		}
 	}
+
+	return nil
+}
+
+// validateStaticIPReservations - validate static IP address reservations
+func (r *OpenStackNetConfig) validateStaticIPReservations(old runtime.Object) error {
```

Comment:
the ``old`` arg looks not used in this func?

Reply:
I added it as I want to extend the func to validate on update that there are no dupes on static and dynamic created ips. will enhance that now